### PR TITLE
add contributing doc, drop redundant readme info

### PR DESCRIPTION
**Pull request by infinitewarp**
https://github.com/cloudigrade/cloudigrade/issues/145

### CODE REVIEW

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+
+When you submit your pull request, include a link in the description to the issue that the code change is addressing. Pull requests must be assigned to the **milestone** for the current sprint by a maintainer before we accept it. With ZenHub, pull requests should also be connected to the relevant issue.
+
+**cloudigrade** has enabled the following third-party integrations that must complete for any pull request:
+
+- `Travis CI <https://travis-ci.org/>`_ for running Flake8 and tests and building artifacts
+- `codecov <https://codecov.io/>`_ for tracking and enforcing tested code coverage
+- `pyup <https://pyup.io/>`_ for checking dependency versions
+
+If you are creating a pull request from a personal fork of the repo, you will need to grant access to your GitHub account for all of those services. Maintainers of **cloudigrade** are encouraged to clone the main repo instead of creating personal forks in the interest of simplifying configuration.
```
[the remark this answers is not in the document]
Comment (infinitewarp) [reply CONTRIBUTING.rst]: OK, I'll remove that. Out of curiosity, what was the problem that caused things to break for Brett the other day?

[the remark this answers is not in the document]
Comment (ilyavw) [reply CONTRIBUTING.rst]: We have two main drivers for all of our CI, travis and pyup. With Travis we had PR builds disabled, thats the only type of build you get in a PR from a Fork. The other one, pyup, by default sets up its webhook to only fire on pushes, normally fine, but same way, if its a fork it never technically pushes a commit in our repo, so I had to update that to fire also on PR creation/modification.

File: README.md [hunk 2]
```diff
@@ -8,134 +8,25 @@
...
-![](docs/pycharm-settings-imports.png)
-
-Alternatively, you may use the command-line tool [isort](https://pypi.python.org/pypi/isort) which has default settings that match closely enough for cloudigrade. `isort` can be used to automatically clean up imports across many files, but please manually review its changes before committing to ensure that there are no unintended side-effects. Example usage:
-
-    # view diff of suggested changes
-    isort -df -rc ./cloudigrade/
-
-    # apply all changes to files
-    isort -rc ./cloudigrade/
+We do not yet have concice setup notes for running **cloudigrade**, and we currently require setting up a complete development envirionment. Watch this space for changes in the future, but for now, please read: [Setting up a development environment](./CONTRIBUTING.rst#setting-up-a-development-environment)
```
Comment (kdelee) [inline README.md]: `s/concice/concise/`

Comment (infinitewarp) [reply README.md]: 😱

File: CONTRIBUTING.rst [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+Contributing code
+=================
+
+**cloudigrade** code lives on `GitHib <https://github.com/cloudigrade/>`_, and all contributions should be submitted there via pull requests.
+
+
+Branching strategy
+------------------
+
+**cloudigrade** follows a simplified `git flow <http://nvie.com/posts/a-successful-git-branching-model/>`_. The ``master`` branch is production-like and reflects the state of the released/live running service at any time (thanks to continuous deployment). All in-development work lives in other branches. We do *not* have perpetual ``develop`` or ``release`` branches. Changes are introduced to master through pull requests directly from short-lived feature branches.
```
Comment (adberglund) [inline CONTRIBUTING.rst]: It's almost github flow (https://guides.github.com/introduction/flow/) except that we don't currently "deploy" our branches.

**Comment by infinitewarp** [comment]
@blentz I think that's a good point, and as I was writing this, I was actually waffling on where "dev environment" should go. I've shuffled the contents of the dev environment section back to the README, and I also took this opportunity to reformat the README to rst.

**Pull request merged**